File: apps/motoboy/app/src/main/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/DeliveryStatusLabels.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

/** Traduz o status bruto da API para o texto exibido ao motoboy. */
fun deliveryStatusLabel(status: String): String {
  return DELIVERY_STATUS_LABELS[status] ?: fallbackStatusLabel(status)
}

/** Mantem status novos legiveis enquanto o app ainda nao conhece o enum. */
private fun fallbackStatusLabel(status: String): String {
  val normalized = status.trim().replace("_", " ").lowercase()
  if (normalized.isBlank()) return "Status desconhecido"
  return normalized.replaceFirstChar { it.uppercase() }
}

private val DELIVERY_STATUS_LABELS = mapOf(
  "RASCUNHO" to "Rascunho",
  "AGUARDANDO_MOTOBOY" to "Disponivel",
  "ACEITA_PELO_MOTOBOY" to "Aceita",
  "COLETADA" to "Coletada",
  "EM_ROTA" to "Em rota",
  "ENTREGUE" to "Entregue",
  "PROBLEMA" to "Problema",
  "CANCELADA" to "Cancelada",
)

File: apps/motoboy-pwa/src/deliveryStatusLabels.ts
const DELIVERY_STATUS_LABELS: Record<string, string> = {
  RASCUNHO: "Rascunho",
  AGUARDANDO_MOTOBOY: "Disponivel",
  ACEITA_PELO_MOTOBOY: "Aceita",
  COLETADA: "Coletada",
  EM_ROTA: "Em rota",
  ENTREGUE: "Entregue",
  PROBLEMA: "Problema",
  CANCELADA: "Cancelada",
};

export function deliveryStatusLabel(status: string): string {
  return DELIVERY_STATUS_LABELS[status] ?? fallbackStatusLabel(status);
}

function fallbackStatusLabel(status: string): string {
  const normalized = status.trim().replaceAll("_", " ").toLowerCase();
  if (!normalized) return "Status desconhecido";
  return normalized.charAt(0).toUpperCase() + normalized.slice(1);
}
